src/core/sysfs.cc

```diff
@@ -99,7 +100,7 @@ static string sysfs_getbustype(const string & path)
   {
     devname =
       string(fs.path + "/bus/") + string(namelist[i]->d_name) +
-      "/devices/" + basename(path.c_str());
+      "/devices/" + basename(const_cast<char*>(path.c_str()));
```

Comment:
this is wrong -- when you include `libgen.h`, you get a POSIX `basename` which is allowed to modify the memory of the arg passed in.  which means this might crash.

might be better to have lshw define its own basename to guarantee portability.